test/e2e/apps/disruption.go

```diff
@@ -215,6 +201,38 @@ var _ = SIGDescribe("DisruptionController", func() {
 			}
 		})
 	}
+
+	var _ = Describe("UpdatingPodDisruptionBudgets", func() {
+		It("Allows updates to pdb's", func() {
+			By("Creating a pdb that targets all three pods in a test replica set")
+			createPDBMinAvailableOrDie(cs, ns, intstr.FromInt(3))
+			createReplicaSetOrDie(cs, ns, 3, false)
+			waitForPodsOrDie(cs, ns, 3)
+
+			By("First trying to evict a pod which shouldn't be evictable")
+			var pod v1.Pod
+			pod, err := locateARunningPod(cs, ns)
+			Expect(err).NotTo(HaveOccurred())
+
+			e := &policy.Eviction{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      pod.Name,
+					Namespace: ns,
+				},
+			}
+			time.Sleep(timeout)
+			err = cs.CoreV1().Pods(ns).Evict(e)
+			Expect(err).Should(MatchError("Cannot evict pod as it would violate the pod's disruption budget."))
+
+			By("Updating the pdb to allow a pod to be evicted")
+			updatePDBMinAvailableOrDie(cs, ns, intstr.FromInt(2))
+			time.Sleep(timeout)
```

Comment:
Like the case in the table-based test where `c.shouldDeny` is `false`, we can speed this up by retrying the eviction instead of using a big, fixed sleep:

https://sourcegraph.com/github.com/kubernetes/kubernetes@7693a1d5fe2a35b6e2e205f03ae9b3eddcdabc6b/-/blob/test/e2e/apps/disruption.go#L204-214

Reply:
ok, have taken out the waits and retry the eviction after checking we have the correct number of pods running